rdt/transformers/base.py

```diff
@@ -9,20 +9,118 @@ class BaseTransformer:
     and ``fit_transform`` method is already implemented.
     """
 
-    def fit(self, data):
+    INPUT_TYPE = None
+    OUTPUT_TYPES = None
+    DETERMINISTIC_TRANSFORM = None
+    DETERMINISTIC_REVERSE = None
+    COMPOSITION_IS_IDENTITY = None
+    NEXT_TRANSFORMERS = None
+
+    def get_input_type(self):
+        """Return the input type supported by the transformer.
+
+        Returns:
+            string:
+                Accepted input type of the transformer.
+        """
+        return self.INPUT_TYPE
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        return self.OUTPUT_TYPES
```

Comment:
I implemented it, but I think this has a few implications:

1.  First, this means we have to append something at the end of the column name always, even if only one column was generated, instead of not appending anything.
2. Currently, since we apply the null_transformer at the end of most transformers, this logic won't be used, since we have to check if the data has `None`'s, which means we have to overwrite this function.